sky/packages/sky/lib/widgets/date_picker.dart

```diff
@@ -0,0 +1,392 @@
+      DateTime.THURSDAY,
+      DateTime.FRIDAY,
+      DateTime.SATURDAY
+    ];
+    int daySlots = weeksShown * days.length;
+    List<Widget> labels = [];
+    for (int i = 0; i < daySlots; i++) {
+      // This assumes a start day of SUNDAY, but could be changed.
+      int day = i - firstDay + 1;
+      Widget item;
+      if (day < 1 || day > daysInMonth) {
+        item = new Text("");
+      } else {
+        // Put a light circle around the selected day
+        BoxDecoration decoration = null;
+        if (selectedDate.year == year &&
+            selectedDate.month == month &&
+            selectedDate.day == day)
+          decoration = new BoxDecoration(
+            backgroundColor: theme.primarySwatch[100],
+            shape: Shape.circle
+          );
+
+        // Use a different font color for the current day
+        TextStyle itemStyle = dayStyle;
+        if (currentDate.year == year &&
+            currentDate.month == month &&
+            currentDate.day == day)
+          itemStyle = itemStyle.copyWith(color: theme.primaryColor);
+
+        item = new Listener(
+          onGestureTap: (_) {
+            DateTime result = new DateTime(year, month, day);
+            if (onChanged != null)
+              onChanged(result);
+          },
+          child: new Container(
+            height: 30.0,
+            decoration: decoration,
+            child: new Center(
+              child: new Text(day.toString(), style: itemStyle)
+            )
+          )
+        );
+      }
+      labels.add(new Flexible(child: item));
+    }
+    for (int w = 0; w < weeksShown; w++) {
+      int startIndex = w * days.length;
+      rows.add(new Container(
+        child: new Flex(
+          labels.sublist(startIndex, startIndex + days.length),
+          justifyContent: FlexJustifyContent.spaceAround
+        )
+      ));
+    }
+
+    return new Column(rows);
+  }
+}
+
+// Scrollable list of DayPickers to allow choosing a month
+class MonthPicker extends ScrollableWidgetList {
+  MonthPicker({
+    this.selectedDate,
+    this.onChanged,
+    this.firstDate,
+    this.lastDate,
+    double itemExtent
+  }) : super(itemExtent: itemExtent) {
+    assert(selectedDate != null);
+    assert(lastDate.isAfter(firstDate));
+  }
+  DateTime selectedDate;
+  DatePickerValueChanged onChanged;
+  DateTime firstDate;
+  DateTime lastDate;
+
+  void syncConstructorArguments(MonthPicker source) {
+    selectedDate = source.selectedDate;
+    onChanged = source.onChanged;
+    firstDate = source.firstDate;
+    lastDate = source.lastDate;
+    super.syncConstructorArguments(source);
+  }
+
+  @override
+  void initState() {
+    _updateCurrentDate();
+    super.initState();
+  }
+
+  DateTime _currentDate;
+  void _updateCurrentDate() {
+    _currentDate = new DateTime.now();
+    DateTime tomorrow = new DateTime(_currentDate.year, _currentDate.month, _currentDate.day + 1);
+    Duration timeUntilTomorrow = tomorrow.difference(_currentDate);
```

Comment:
Maybe wait 1 second into tomorrow so that we don't miss it by rounding or something.